app/Services/Caches/MapMarkerCacheService.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Services\Caches;
+
+use App\Models\MapMarker;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\DB;
+
+class MapMarkerCacheService extends BaseCache
+{
+    /**
+     * @return array
+     */
+    public function iconSuggestion(): array
+    {
+        $key = $this->iconSuggestionKey();
+        if (Cache::has($key)) {
+            return Cache::get($key);
+        }
+
+        $default = [
+            'ra ra-tower',
+            'fa-solid fa-home',
+            'ra ra-capitol',
+            'ra ra-skull',
+            'fa-solid fa-coins',
+            'ra ra-beer',
+            'fa-solid fa-map-marker-alt',
+            'fa-solid fa-thumbtack',
+            'ra ra-wooden-sign',
+            'fa-solid fa-map-pin'
+        ];
+
+        $data = MapMarker::select(DB::raw('custom_icon, MAX(created_at) as cmat'))
```

Comment:
This is building stats on the whole table, not on just the map markers of the campaign